common/sagas/swap/orders.ts

```diff
@@ -97,12 +161,59 @@ export function* postBityOrderCreate(action: BityOrderCreateRequestedSwapAction)
   }
 }
 
+export function* postShapeshiftOrderCreate(
+  action: ShapeshiftOrderCreateRequestedSwapAction
+): SagaIterator {
+  const payload = action.payload;
+  try {
+    yield put(stopLoadShapeshiftRatesSwap());
+    const order = yield apply(shapeshift, shapeshift.sendAmount, [
+      payload.withdrawal,
+      payload.originKind,
+      payload.destinationKind,
+      payload.destinationAmount
+    ]);
+    if (order.error) {
+      yield put(showNotification('danger', `Shapeshift Error: ${order.error}`, TEN_SECONDS));
+      yield put(shapeshiftOrderCreateFailedSwap());
+    } else {
+      yield put(shapeshiftOrderCreateSucceededSwap(order.success));
+      yield put(changeStepSwap(3));
+      // start countdown
+      yield put(startOrderTimerSwap());
+      // start shapeshift order status polling
+      yield put(startPollShapeshiftOrderStatus());
+    }
+  } catch (e) {
+    const message =
+      'Connection Error. Please check the developer console for more details and/or contact support';
+    yield put(showNotification('danger', message, TEN_SECONDS));
+    yield put(shapeshiftOrderCreateFailedSwap());
+  }
+}
+
 export function* postBityOrderSaga(): SagaIterator {
-  yield takeEvery('SWAP_ORDER_CREATE_REQUESTED', postBityOrderCreate);
+  yield takeEvery(TypeKeys.SWAP_BITY_ORDER_CREATE_REQUESTED, postBityOrderCreate);
+}
+
+export function* postShapeshiftOrderSaga(): SagaIterator {
+  yield takeEvery(TypeKeys.SWAP_SHAPESHIFT_ORDER_CREATE_REQUESTED, postShapeshiftOrderCreate);
+}
+
+export function* restartSwap() {
+  yield put(reset());
+  yield put(resetWallet());
+  yield put(stopPollShapeshiftOrderStatus());
+  yield put(stopPollBityOrderStatus());
+  yield put(loadShapeshiftRatesRequestedSwap());
+}
+
+export function* restartSwapSaga(): SagaIterator {
+  yield takeEvery(TypeKeys.SWAP_RESTART, restartSwap);
 }
 
-export function* bityTimeRemaining(): SagaIterator {
-  while (yield take('SWAP_ORDER_START_TIMER')) {
+export function* orderTimeRemaining(): SagaIterator {
```

Comment:
This saga has turned into a monster. Can we break this out into separate generators?